benchmark/primitives/primitives_test.go

```diff
@@ -188,6 +189,59 @@ func BenchmarkMutexWithoutDefer(b *testing.B) {
 	}
 }
 
+func BenchmarkIncrInt64Atomics(b *testing.B) {
+	var c int64
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		atomic.AddInt64(&c, 1)
+	}
+	b.StopTimer()
+	if c != int64(b.N) {
+		b.Fatal("error")
+	}
+}
+
+func BenchmarkIncrInt64Mutex(b *testing.B) {
+	var c int64
+	mu := sync.Mutex{}
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		mu.Lock()
+		c++
+		mu.Unlock()
+	}
+	b.StopTimer()
+	if c != int64(b.N) {
+		b.Fatal("error")
+	}
+}
+
+func BenchmarkSetTimeAtomics(b *testing.B) {
```

Comment:
We have some existing benchmarks of atomics above.  Can you name this `BenchmarkAtomicValueStore` (and rename `BenchmarkAtomicValue` to `BenchmarkAtomicValueLoad`) to match?  Also, the value type used should be the same -- if a `time.Time` isn't important, I'd rather use an `int` instead.

Reply:
From benchmark results, time.Time (~100ns) takes twice the time of int (~50ns).